### PR TITLE
Add caching stats & tests

### PR DESCRIPTION
**Pull request by josvazg**
This could be an alternative to #118 
It includes those same tests, but simplified as the new stats allow for some simpler checks that the caching is actually working as expected end to end.

### CODE REVIEW

File: test/external/external_test.go [hunk 1]
```diff
@@ -220,17 +241,37 @@ var _ = Describe("External tests", func() {
 			Eventually(test.CommandSession.Out, time.Minute).Should(Say("Writing 2 images...\n"))
 		})
 
+		define.Then(`^the command says local cache saved 0 bytes in downloads$`, func() {
+			Eventually(test.CommandSession.Out, time.Minute).Should(Say("Local cache saved 0 out of 767322 bytes in layer downloads\n"))
```
Comment (migmartri) [inline test/external/external_test.go:245]: I don't like that our tests depend on log output, couldn't we read from the struct that we are storing?

**Review by migmartri — CHANGES_REQUESTED** [review]

**Comment by migmartri** [comment]
I am rejecting this patch because it adds additional, irrelevant output noise to the user when it's more like a debugging property.

File: pkg/mover/intermediate.go [hunk 2]
```diff
@@ -123,21 +123,35 @@ func tarImages(imageChanges []*internal.ImageChange, cacheDir string, logger Log
...
 }
 
+func stats(logger Logger, cachedImages []*internal.CachedImage) {
+	var cached, total int64
+	for _, cachedImage := range cachedImages {
+		stats := cachedImage.Stats()
+		cached += stats.CachedBytes
+		total += stats.DownloadedBytes + stats.CachedBytes
+	}
+	logger.Printf("Local cache saved %d out of %d bytes in layer downloads\n", cached, total)
```
[the remark this answers is not in the document]
Comment (josvazg) [reply pkg/mover/intermediate.go:152]: Could I pass a env var to get this output and use it here?

Doing proper deep testing is not something I have time for now and this is much better than not having testing on such a feature at all. I believe this covers any regression that might arise in caching.

I do agree that for regular users caching is irrelevant, but for advanced users or operators doing heavy use of reloks, specially under chat-syncer it is useful to see or debug how caching is helping or not.

File: test/external/external_test.go (hunk 1 above)
Comment (josvazg) [reply test/external/external_test.go:245]: All this external tests do test depending on output. All of them.

**Comment by josvazg** [comment]
Due to current complaints about this change, it seems better to closet this and instead resurrect #118 so that we have some testing coverage without adding cache stats output.

**Pull request closed without merging**